src/printer.cpp

```diff
@@ -43,27 +43,29 @@ namespace libcellml {
  */
 struct Printer::PrinterImpl
 {
-    std::string printUnits(const UnitsPtr &units) const;
-    std::string printComponent(const ComponentPtr &component) const;
-    std::string printEncapsulation(const ComponentPtr &component) const;
-    std::string printVariable(const VariablePtr &variable) const;
-    std::string printReset(const ResetPtr &reset) const;
-    std::string printResetChild(const std::string &childLabel, const std::string &childId, const std::string &math) const;
+    std::string printUnits(const UnitsPtr &units, IdList &idList, bool autoIds);
+    std::string printComponent(const ComponentPtr &component, IdList &idList, bool autoIds);
+    std::string printEncapsulation(const ComponentPtr &component, IdList &idList, bool autoIds);
+    std::string printVariable(const VariablePtr &variable, IdList &idList, bool autoIds);
+    std::string printReset(const ResetPtr &reset, IdList &idList, bool autoIds);
+    std::string printResetChild(const std::string &childLabel, const std::string &childId, const std::string &math, IdList &idList, bool autoIds);
```

Comment:
Why have these lost there constness?

Reply:
Ho hum.  The new argument (`idList`) changes, but it's not part of the class.  I guess it should be const then.  But do you want it on the bools too? I'm confused by your later comment @hsorby ?

Reply:
Can't change the idList, sorry.  I've removed the const bools though. 23a39d46